contrib/hashicorp/vault/vault.go

```diff
@@ -66,7 +67,8 @@ func WrapHTTPClient(c *http.Client, opts ...Option) *http.Client {
 				return
 			}
 			s.SetTag(ext.HTTPCode, res.StatusCode)
-			if res.StatusCode >= 400 {
+			// treat 4XX and 5XX as errors for a client
+			if globalconfig.IsHTTPClientError(res.StatusCode) || globalconfig.IsHTTPServerError(res.StatusCode) {
```

Comment:
Is this the right thing to do?

Reply:
This is logically the same as what was there before, but I think the previous check was a mistake.